app/controllers/movies_controller.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+class MoviesController < ApplicationController
+  before_action :set_movie, only: %i[show update destroy]
+
+  # GET /movies
+  def index
+    @movies = Movie.all
```

Comment:
It's not neccessary to create an instance variable. A simple local variable would be enough. 
This approach makes sense when the view layer is added to your application but since we do simple API it's redundant.

Reply:
so I should replace all instance variables in all controllers into local variables? I'm not sure if I understand Ruby's variables differences

Reply:
Ok so basically it's redundant in the index method because you are using this just in the scope of this method. 
I see that you're using the before_filter approach to fetch your movie before trigger the actual `show` method. In such a condition, you need to make it an instance variable to be visible/fetchable around your controller class. 
Additionally, this approach is also not very practical because I need to travel around your class up and down to understand the flow. See [Link_1](https://thoughtbot.com/blog/before-filter-wisdom) and [Here](https://craftingruby.com/posts/2015/05/31/dont-use-before-action-to-load-data.html)

Reply:
Additionally here https://stackoverflow.com/a/14319518

Reply:
nevermind, @wmaciejak explained the difference :)

Reply:
I haven't seen your newest answers. Thanks, now I get it :)